components/aws/sagemaker/model/src/sagemaker_model_spec.py

```diff
@@ -0,0 +1,145 @@
+    SageMakerComponentSpec[SageMakerCreateModelInputs, SageMakerCreateModelOutputs]
+):
+    INPUTS: SageMakerCreateModelInputs = SageMakerCreateModelInputs(
+        model_name=InputValidator(
+            input_type=str, required=True, description="The name of the new model."
+        ),
+        role=InputValidator(
+            input_type=str,
+            required=True,
+            description="The Amazon Resource Name (ARN) that Amazon SageMaker assumes to perform tasks on your behalf.",
+        ),
+        container_host_name=InputValidator(
+            input_type=str,
+            required=False,
+            description="When a ContainerDefinition is part of an inference pipeline, this value uniquely identifies the container for the purposes of logging and metrics.",
+            default="",
+        ),
+        image=InputValidator(
+            input_type=str,
+            required=False,
+            description="The Amazon EC2 Container Registry (Amazon ECR) path where inference code is stored.",
+            default="",
+        ),
+        model_artifact_url=InputValidator(
+            input_type=str,
+            required=False,
+            description="S3 path where Amazon SageMaker to store the model artifacts.",
+            default="",
+        ),
+        environment=InputValidator(
+            input_type=SpecInputParsers.yaml_or_json_dict,
+            required=False,
+            description="The dictionary of the environment variables to set in the Docker container. Up to 16 key-value entries in the map.",
+            default={},
+        ),
+        model_package=InputValidator(
+            input_type=str,
+            required=False,
+            description="The name or Amazon Resource Name (ARN) of the model package to use to create the model.",
+            default="",
+        ),
+        secondary_containers=InputValidator(
+            input_type=SpecInputParsers.yaml_or_json_dict,
+            required=False,
+            description="A list of dicts that specifies the additional containers in the inference pipeline.",
+            default={},
+        ),
+        vpc_security_group_ids=InputValidator(
+            input_type=str,
+            required=False,
+            description="The VPC security group IDs, in the form sg-xxxxxxxx.",
+            default="",
+        ),
+        vpc_subnets=InputValidator(
+            input_type=str,
+            required=False,
+            description="The ID of the subnets in the VPC to which you want to connect your hpo job.",
+            default="",
+        ),
+        network_isolation=InputValidator(
+            input_type=SpecInputParsers.str_to_bool,
+            required=False,
+            description="Isolates the training container.",
+            default=True,
+        ),
+        **vars(COMMON_INPUTS),
+    )
+
+    OUTPUTS = SageMakerCreateModelOutputs(
+        model_name=OutputValidator(description="The model name SageMaker created."),
```

Comment:
Sentence structure is bit confusing 
maybe change this to "Name of the Model created on SageMaker"
or just "The SageMaker Model name"